src/adjusthandler.cpp

```diff
@@ -14,21 +15,43 @@
 #endif
 
 #include <QString>
+#include <QRandomGenerator>
 
-void AdjustHandler::initialize(quint16 proxyPort) {
-  if (!AdjustProxy::instance()->isListening()) {
+namespace {
+bool s_initialized = false;
+}  // namespace
+
+void AdjustHandler::maybeInitialize() {
+  if (s_initialized) {
+    return;
+  }
+
```

Comment:
Move `m_initialized = true` here.

Reply:
In this way, if the proxy doesn't start, we do not try each time we change state.